### PR TITLE
Task 2 : Kenny McDonnell

### PR DESCRIPTION
**Pull request by KennyMcD**

### CODE REVIEW

File: starter/src/test/java/com/starter/fullstack/rest/InventoryControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    this.inventory.setName("ALSO TEST");
+    this.mockMvc.perform(post("/inventory")
+        .accept(MediaType.APPLICATION_JSON)
+        .contentType(MediaType.APPLICATION_JSON)
+        .content(this.objectMapper.writeValueAsString(this.inventory)))
+      .andExpect(status().isOk());
+
+    List<Inventory> inv_list = this.mongoTemplate.findAll(Inventory.class);
+    
+    Assert.assertNotEquals("OTHER ID", inv_list.get(1).getId());
```
Comment (jdiangelo) [inline starter/src/test/java/com/starter/fullstack/rest/InventoryControllerTest.java]: Line 66 and 68 are not really necessary, the DAO test checks the ids, so you don't need to do it here too. Everything else here looks great!

File: starter/src/main/java/com/starter/fullstack/rest/InventoryController.java [hunk 2]
```diff
@@ -25,6 +27,16 @@ public InventoryController(InventoryDAO inventoryDAO) {
     this.inventoryDAO = inventoryDAO;
   }
 
+  /**
+   * Save Inventory.
+   * @param inventory Inventory to Save/Update.
+   * @return Created/Updated Inventory.
+   */
+  @PostMapping
+  public Inventory create(@RequestBody Inventory inventory) {
```
Comment (jdiangelo) [inline starter/src/main/java/com/starter/fullstack/rest/InventoryController.java]: How can we check if the data that you send to the controller method is valid or not?

**Comment by jdiangelo** [comment]
Please answer the following Knowledge Check Questions in 1-3 sentences each:
1. What’s the purpose of the RequestMapping on line 17? 
    Request Mapping is used to map HTTP requests to Spring controller methods.
2. What is mockMvc doing in the J-Unit test?
    MockMvc allows us to test the web layer of the full stack without the need to start up a server. Spring handles the HTTP request and hands it off to the controller to test.
3. How are we passing values to the controller’s create method? 
    Through the PostMapping annotation we map HTTP POST requests to the create controller method. Using mockMvc, we execute a post request with the content attribute set to an inventory object.

File: starter/src/main/java/com/starter/fullstack/rest/InventoryController.java (hunk 2 above)
Comment (jdiangelo) [reply starter/src/main/java/com/starter/fullstack/rest/InventoryController.java]: You can use the @Valid annotation next to where @RequestBody is to ensure validation of the whole object.

Comment (KennyMcD) [reply starter/src/main/java/com/starter/fullstack/rest/InventoryController.java]: Added the valid tag, my create test was failing at first. Turns out Product Type can't be null for the inventory to be valid.

**Pull request merged**